os/kernel/task/task_prctl.c

```diff
@@ -133,15 +133,9 @@ int prctl(int option, ...)
 		int pid = va_arg(ap, int);
 		FAR struct tcb_s *tcb;
 
-		/* Get the TCB associated with the PID (handling the special case of
-		 * pid==0 meaning "this thread")
-		 */
+		/* Get the TCB associated with the PID */
 
-		if (!pid) {
-			tcb = this_task();
-		} else {
-			tcb = sched_gettcb(pid);
-		}
+		tcb = sched_gettcb(pid);
```

Comment:
I think we'd better keep original to get/set name of calling thread at once without pid by calling getpid().

Reply:
Then, how about using `MAX_PID_MASK`? If arg3 is MAX_PID_MASK, then it uses `this_task` to get current running task name.

Reply:
@jeongarmy , As I mentioned in the commit msg, it is better to use its own pid.
@sunghan-chang , using `MAX_PID_MASK` is not good for understanding, I think.